view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -56,13 +56,110 @@ define(
                 return this;
             },
 
+            /**
+             * Get Installment minimum
+             * this function respects info from: https://www.omise.co/installment-payment
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @return {integer}
+             */
+            getInstallmentMinimum(id) {
+                switch (id) {
+                    case 'kbank':        return 500;
+                    case 'bbl':          return 500;
+                    case 'bay':          return 300;
+                    case 'first_choice': return 300;
+                    case 'ktc':          return 300;
+                }
+            },
+
+            /**
+             * Get Installment monthly interest rate
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @return {float}
+             */
+            getinstallmentInterestRate(id) {
+                switch (id) {
+                    case 'kbank':        return 0.0065;
+                    case 'bbl':          return 0.008;
+                    case 'bay':          return 0.008;
+                    case 'first_choice': return 0.013;
+                    case 'ktc':          return 0.008;
+                }
+            },
+
+            /**
+             * Get VAT tax rate
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @return {float}
+             */
+            getVatRate() {
+                return 0.07;
+            },
+
+            /**
+             * Get zero interest setting
+             *
+             * @return {string}
+             */
+            isZeroInterest() {
+                return window.checkoutConfig.is_zero_interest;
+            },
+
+            /**
+             * Calculates single installment amount
+             *
+             * @return {integer}
+             */
+            calculateSingleInstallmentAmount(id, terms) {
+                const total = this.getTotal();
+                const tax = this.getVatRate();
+
+                if (this.isZeroInterest()) {
+                    //merchant pays interest
+                    return (total / terms).toFixed(0)
+                }
+
+                // get monthly interest setting
+                const rate = this.getinstallmentInterestRate(id);
+                const interest = ((1 + tax) * rate * terms * total);
+
+                return + ( ( (total + interest ) / terms ).toFixed( 0 ) );
```

Comment:
@jacstn this one I forgot to test, but wonder why we want to round up those decimal? Why not leave it as `toFixed(2)` or there is no case that this would be 2 digits decimal?

Reply:
we round up because the minimum is 300 THB, so in case of such a big number showing decimals does not matter (I think), it is only to show people how much more or less they will pay, the rest they will see on bank pages.